packages/manager/modules/pci/src/projects/project/storages/databases/database.service.js

```diff
@@ -0,0 +1,351 @@
+import filter from 'lodash/filter';
+import find from 'lodash/find';
+import get from 'lodash/get';
+import map from 'lodash/map';
+import set from 'lodash/set';
+import 'moment';
+
+import Backup from '../../../../components/project/storages/databases/backup.class';
+import Database from '../../../../components/project/storages/databases/database.class';
+import Engine from '../../../../components/project/storages/databases/engine.class';
+
+export default class DatabaseService {
+  /* @ngInject */
+  constructor($http, $q, $translate, CucPriceHelper) {
+    this.$http = $http;
+    this.$q = $q;
+    this.$translate = $translate;
+    this.CucPriceHelper = CucPriceHelper;
+  }
+
+  static getAPIRoot(projectId, engine) {
+    return `/cloud/project/${projectId}/database/${engine}`;
+  }
+
+  static getDatabaseBaseUrl(projectId, engine, databaseId) {
+    return `${DatabaseService.getAPIRoot(projectId, engine)}/${databaseId}`;
+  }
+
+  static getIcebergHeaders() {
+    return {
+      headers: {
+        'X-Pagination-Mode': 'CachedObjectList-Pages',
+        'X-Pagination-Size': 50000,
+      },
+    };
+  }
+
+  addNode(projectId, engine, databaseId, flavor, region) {
+    return this.$http
+      .post(
+        `${DatabaseService.getDatabaseBaseUrl(
```

Comment:
Not entirely sure what are the benefits here, first impression is it's making it harder to read and to find which API is called

Reply:
Actually, the only benefit is that its easier to change the base API url. But I have reverted this change back